core/domain/opportunity_services.py

```diff
@@ -0,0 +1,380 @@
+        model.assigned_voice_artist_in_languages)
+
+
+def _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list):
+    """Stores multiple ExplorationOpportunitySummary into datastore as a
+    ExplorationOpportunitySummaryModel.
+
+    Args:
+        exploration_opportunity_summary_list: list(
+            ExplorationOpportunitySummary). A list of exploration opportunity
+            summary object.
+    """
+    exploration_opportunity_summary_model_list = []
+    for opportunity_summary in exploration_opportunity_summary_list:
+        model = opportunity_models.ExplorationOpportunitySummaryModel(
+            id=opportunity_summary.id,
+            topic_id=opportunity_summary.topic_id,
+            topic_name=opportunity_summary.topic_name,
+            story_id=opportunity_summary.story_id,
+            story_title=opportunity_summary.story_title,
+            chapter_title=opportunity_summary.chapter_title,
+            content_count=opportunity_summary.content_count,
+            incomplete_translation_languages=(
+                opportunity_summary.incomplete_translation_languages),
+            translation_counts=opportunity_summary.translation_counts,
+            need_voice_artist_in_languages=(
+                opportunity_summary.need_voice_artist_in_languages),
+            assigned_voice_artist_in_languages=(
+                opportunity_summary.assigned_voice_artist_in_languages)
+        )
+
+        exploration_opportunity_summary_model_list.append(model)
+
+    opportunity_models.ExplorationOpportunitySummaryModel.put_multi(
+        exploration_opportunity_summary_model_list)
+
+
+def add_new_exploration_opportunities(story, exp_ids):
+    """Adds new exploration opportunity into the model.
+
+    Args:
+        story: Story. A story object.
+        exp_ids: list(str). A list of exploration ids for which new
+            opportunities are to be created.
+    """
+    topic = topic_services.get_topic_by_id(story.corresponding_topic_id)
+    explorations = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+
+    exploration_opportunity_summary_list = []
+    for exp_id, exploration in explorations.iteritems():
+        node = story.story_contents.get_node_with_corresponding_exp_id(exp_id)
+
+        audio_language_codes = set([
+            language['id'] for language in constants.SUPPORTED_AUDIO_LANGUAGES])
+
+        complete_translation_languages = set(
+            exploration.get_languages_with_complete_translation())
+
+        incomplete_translation_languages = (
+            audio_language_codes - complete_translation_languages)
+        need_voice_artist_in_languages = complete_translation_languages
+
+        if exploration.language_code in incomplete_translation_languages:
+            # Removing exploration language from incomplete translation
+            # languages list as exploration does not need any translation in
+            # it's own language.
```

Comment:
it's --> its, here and elsewhere.

("it's" == "it is", which is not what you mean here.)